simulation/traffic_simulator/src/entity/entity_base.cpp

```diff
@@ -818,5 +818,32 @@ auto EntityBase::updateTraveledDistance(const double step_time) -> double
   return traveled_distance_;
 }
 
+auto EntityBase::getDistanceToSyncronizationPosition(
+  const lanelet::Id &
+    intersection_id           // dont know about this, maybe i can get the other infomation?
+  const double stop_position  // may be if the info givven is only the intersection, perhaps needs
+                              // info about how far the stop position is from the intersection
+  ) -> double
+{
+  // return distance to the nearest intersection on the route from ego
+  return entity_distance_to_syncronization_position_;
+}
+
+auto EntityBase::adjustElementVelocityByIntersection(
+  const double ego_distance_to_intersection, const double max_acceleration, const double max_jerk, )
+{
+  double calculated_velocity = 0.0;
+  // get ego info inside the function
+  // adjusting velocity of the element by the distance to the intersection
+  requestSpeedChange(calculated_velocity, false);
+}
```

Comment:
This function is only used in SynchronizeAction. It is recommended to implement it with a lambda expression.